src/modules/Expanders.jl

```diff
@@ -261,17 +261,75 @@ end
 # @docs
 # -----
 
+immutable MethodNode
+    method  :: Method
+    visible :: Bool
+end
+
 immutable DocsNode
-    docstr :: Any
-    anchor :: Anchors.Anchor
-    object :: Utilities.Object
-    page   :: Documents.Page
+    docstr  :: Any
+    anchor  :: Anchors.Anchor
+    object  :: Utilities.Object
+    page    :: Documents.Page
+    """
+    Vector of methods associated with this `DocsNode`. Being nulled means that
+    conceptually the `DocsNode` has no table of method (as opposed to having
+    an empty table).
+    """
+    methods :: Nullable{Vector{MethodNode}}
 end
 
 immutable DocsNodes
     nodes :: Vector{DocsNode}
 end
 
+"""
+Returns a `Nullable{Vector{MethodNode}}` with the methods associated with the `object`.
+
+Null is returned if the object conceptually does not have a table of methods (e.g.
+modules). If instead the object should have a table of methods which just happens
+to be empty then it returns an empty vector.
+
+The methods are also filtered, with `MethodNode.visible` set to false for those
+methods not defined in this package. This allows the writers to choose how they
+want to treat trivial constructors, functions imported from `Base` and such.
+
+Due to differences between 0.4 and 0.5, only 0.5 is supported currently. On 0.4
+we always drop the methods table.
+"""
+function docsnode_methodlist end
+
+if VERSION > v"0.5-"
+    function docsnode_methodlist(object::Utilities.Object, page, doc)
+        if !isdefined(object.binding.mod, object.binding.var)
+            info("Unable to find $(object)")
+            return Nullable{Vector{MethodNode}}()
+        end
+
+        # We only generate the method table for a Function, Type and Macro.
+        # The only other one with methods should be Method itself (e.g. defined
+        # with `foo(x,y)` in the @docs block), but in that case the user has
+        # already picked a method and we shouldn't generate a single element
+        # table.
+        doccat = Utilities.doccat(object)
+        if doccat == "Function" || doccat == "Type" || doccat == "Macro"
+            ms = map(methods(getfield(object.binding.mod, object.binding.var))) do m
+                # We filter out the methods not defined in this package.
+                # That is we show the method if it is defined in any of the
+                # `doc.user.modules` modules defined in `makedocs`.
+                visible = any(map(mod->Utilities.issubmodule(m.module, mod), doc.user.modules))
```

Comment:
Since `doc.user.modules` is a `Set{Module}` can this just be

```julia
visible  = m.module in doc.user.modules
```

since we store all the submodules there already.